lib/Runtime/Types/RecyclableObject.h

```diff
@@ -242,22 +249,30 @@ namespace Js {
         // which will emit an error.
         static Var DefaultEntryPoint(RecyclableObject* function, CallInfo callInfo, ...);
 
+        BOOL HasItem(uint32 index);
+        BOOL HasProperty(PropertyId propertyId);
+        BOOL GetProperty(Var originalInstance, PropertyId propertyId, Var* value, PropertyValueInfo* info, ScriptContext* requestContext);
+        BOOL GetProperty(Var originalInstance, JavascriptString* propertyNameString, Var* value, PropertyValueInfo* info, ScriptContext* requestContext);
+        BOOL GetPropertyReference(Var originalInstance, PropertyId propertyId, Var* value, PropertyValueInfo* info, ScriptContext* requestContext);
+        BOOL GetItem(Var originalInstance, uint32 index, Var* value, ScriptContext * requestContext);
+        BOOL GetItemReference(Var originalInstance, uint32 index, Var* value, ScriptContext * requestContext);
+
```

Comment:
Are these explicitly not virtual because the intent is that they are not allowed to be overridden?